### PR TITLE
Add GCM unit test

### PR DESCRIPTION
**Pull request by jacob-pascual**

### CODE REVIEW

File: message_test.go [hunk 1]
```diff
@@ -57,4 +59,77 @@ func TestAPNS(t *testing.T) {
...
+
+	mockGCM.AssertCalled(t, "Send", mock.AnythingOfType("*gcm.Message"), mock.AnythingOfType("int"))
+
+	message := mockGCM.Calls[0].Arguments[0].(*gcm.Message)
+
+	if message.RegistrationIDs == nil || len(message.RegistrationIDs) != 2 {
+		t.Fatalf("Expected there to be two registration IDs, instead %+v in %+v", message.RegistrationIDs, message)
+	}
+
+	if message.RegistrationIDs[0] != "123456" {
```
Comment (briankassouf) [inline message_test.go]: Are this and the below tests necessary? they seem to only be testing the mock library

File: server.go [hunk 2]
```diff
@@ -18,6 +19,10 @@ const (
 	pongWait  = 1 * time.Second
 )
 
+type GCMClient interface {
```
Comment (briankassouf) [inline server.go:22]: :+1:

**Comment by briankassouf** [comment]
thanks for killing it on tests!

File: message_test.go (hunk 1 above)
Comment (jacob-pascual) [inline message_test.go]: they are testing that we formed the GCM message correctly. not a huge value test; I'm fine dropping it

**Pull request merged**